src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -632,6 +654,84 @@ private void maybeDelayForTesting()
         }
     }
 
+    private UnfilteredPartitionIterator withQuerySizeTracking(UnfilteredPartitionIterator iterator)
```

Comment:
nit: `withQuerySizeTracking` -> `withResponseSizeTracking`?

Although I might be misunderstanding, in the sense that this isn't really about response size over the wire, but the amount of stuff we have to materialize on-heap (even if we free partitions or even rows as they are consumed?) to create a response...?

Reply:
I didn't go with `withResponseSizeTracking` as the response may be smaller than the size we compute; this blocks on total allocated rather than the final need.

> amount of stuff we have to materialize on-heap
yep, that is mostly it.  Freeing isn't free and is async when GC feels like it, so large materialized reads could cause unstable GCs